app/models/derived_variables/case_log_variables.rb

```diff
@@ -67,8 +68,29 @@ def set_derived_fields!
     self.new_old = new_or_existing_tenant
     self.vacdays = property_vacant_days
 
-    if is_supported_housing? && (scheme && scheme.locations.size == 1)
-      self.location = scheme.locations.first
+    if is_supported_housing? && scheme
+      if scheme.locations.size == 1
+        self.location = scheme.locations.first
+      end
+      if location
+        self.la = location.county
+        self.postcode_full = location.postcode
+        type_of_unit_map = {
+          "Self-contained flat or bedsit" => 1,
+          "Self-contained flat or bedsit with common facilities" => 2,
+          "Shared flat" => 3,
+          "Shared house or hostel" => 4,
+          "Bungalow" => 5,
+          "Self-contained house" => 6,
+        }
+        self.unittype_sh = type_of_unit_map[location.type_of_unit]
+        self.builtype = form.questions.find { |x| x.id == "builtype" }.answer_options.select { |_key, value| value["value"] == location.type_of_building }.keys.first
```

Comment:
Just syntactic sugar but we have `form.get_question("builtype", self)`, probably wants to be a `.find` on the answer options so you stop looping as soon as you have a match.

Reply:
select doesn't fail if it doesn't find anything (that shouldn't be the case, but it is in our tests)

Reply:
Is that a valid scenario?

Reply:
I don't think so (it shouldn't be)

Reply:
Should we change the tests to reflect that then?